app/controllers/documents_controller.rb

```diff
@@ -3,9 +3,26 @@ def download
     doc_download = letter_use_case_factory.download.execute(id: params.fetch(:id))
     doc = doc_download[:document]
 
+    tenancy_ref = get_tenancy_ref(doc)
+    if params[:username] != ''
```

Comment:
I like using the Rails methods ([`blank`](https://apidock.com/rails/Object/blank%3F) and [`present`](https://apidock.com/rails/Object/present%3F)) to make these sorts of conditionals more readable:
```suggestion
    if params[:username].blank?
```

Reply:
`!= ''` is different logic than `#present?`

With `!= ''`
- Whitespace(`' '`) is true
- `nil` is true 
- `'string'` is true
- empty string is false

With `#present?` 
-  whitespace is false 
- `nil` is fale
- `'string'` is true
- empty string is false

Reply:
This is true - I was assuming the intent was to get behaviour like `present?`, but that might not be the case!

Reply:
Yes, ```present?``` works as intended 😄